controllers/volumereplicationgroup_controller.go

```diff
@@ -146,6 +158,53 @@ func (r *VolumeReplicationGroupReconciler) configMapFun(configmap client.Object)
 	return req
 }
 
+func (r *VolumeReplicationGroupReconciler) waitForCrds() error {
+	log := r.Log.WithName("checkingDependencies")
+
+	err := r.waitForResource(log, volumeReplicationClass)
+	if err != nil {
+		log.Error(err, "failed to wait for VolumeReplicationClass CRD")
+
+		return err
+	}
+
+	err = r.waitForResource(log, volumeReplication)
+	if err != nil {
+		log.Error(err, "failed to wait for VolumeReplication CRD")
+
+		return err
+	}
+
+	return nil
+}
+
+func (r *VolumeReplicationGroupReconciler) waitForResource(logger logr.Logger, resourceName string) error {
+	unstructuredResource := &unstructured.UnstructuredList{}
+	unstructuredResource.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   volrep.GroupVersion.Group,
+		Kind:    resourceName,
+		Version: volrep.GroupVersion.Version,
+	})
+
+	for {
+		err := r.Client.List(context.TODO(), unstructuredResource)
+		if err != nil {
+			if !meta.IsNoMatchError(err) {
+				logger.Error(err, "got an unexpected error while waiting for resource", "Resource", resourceName)
+
+				return err
+			}
+		} else {
+			logger.Info("resource found", "Resource", resourceName)
+
+			return nil
+		}
+
+		logger.Info("resource does not exist", "Resource", resourceName)
```

Comment:
If process is not printing and also not proceeding with the initialization - it may be very confusing. I think we'd better see a message every several seconds to be sure process is alive and waiting.

Reply:
This is true for debug level logs - logging one log for every check helps to debug. But for user level logs, the interesting event is starting to wait for crds, and ending the wait. We don't need to log on each check since the system state did not change.

Reply:
Done